js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        let orderbook = undefined;
+        if (market['spot']) {
+            const response = await this.spotPublicGetDepth (this.extend (request, params));
+            //
+            //     {
+            //         "lastUpdateId": "744267132",
+            //         "bids": [
+            //             ["40838.50","0.387864"],
+            //             ["40837.95","0.008400"],
+            //         ],
+            //         "asks": [
+            //             ["40838.61","6.544908"],
+            //             ["40838.88","0.498000"],
+            //         ]
+            //     }
+            //
+            orderbook = this.parseOrderBook (response, symbol);
+            orderbook['nonce'] = this.safeInteger (response, 'lastUpdateId');
+        } else if (market['swap']) {
+            const response = await this.contractPublicGetDepthSymbol (this.extend (request, params));
+            //
+            //     {
+            //         "success":true,
+            //         "code":0,
+            //         "data":{
+            //             "asks":[
+            //                 [3445.72,48379,1],
+            //                 [3445.75,34994,1],
+            //             ],
+            //             "bids":[
+            //                 [3445.55,44081,1],
+            //                 [3445.51,24857,1],
+            //             ],
+            //             "version":2827730444,
+            //             "timestamp":1634117846232
+            //         }
+            //     }
+            //
+            const data = this.safeValue (response, 'data');
+            const timestamp = this.safeInteger (data, 'timestamp');
+            orderbook = this.parseOrderBook (data, symbol, timestamp);
+            orderbook['nonce'] = this.safeInteger (data, 'version');
+        }
+        return orderbook;
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        // if (since !== undefined) {
+        //     request['startTime'] = since; bug in api, waiting for fix
+        // }
+        let trades = undefined;
+        if (market['spot']) {
+            let method = this.safeString (this.options, 'fetchTradesMethod', 'spotPublicGetAggTrades');
```

Comment:
Any reason for making `spotPublicGetAggTrades` the default method? I would say that users are expecting more of what is returned by `spotPublicGetTrades` (regular recent trade info) but I might be missing something here

Reply:
yes, in most cases we use not `/trades`, but `/aggTrades`  endpoint (iirc , you can look through binance's implementation of fetchTrades, and it's same there). the reason behind that is that, getting granular detailed trades required much higher rate limit (i.e. 5x times more) and is not any much different (in binance docs it's explained), as aggTrades satisfies probably 99.9% cases ( and same applies to mexc and other exchanges) .